components/table/Table.razor.cs

```diff
@@ -349,7 +349,15 @@ protected override void OnParametersSet()
             }
         }
 
-        protected override bool ShouldRender() => this._shouldRender;
+        protected override bool ShouldRender()
+        {
+            if (_waitingReload)
+            {
+                _waitingReload = false;
+                Reload();
+            }
```

Comment:
Why not write it in the AfterRender method?

Reply:
It should not reload data in `OnAfterRender` method, the reload action should be happened before render.
@anddrzejb you can move the reload action code from `OnAfterRender` method(`Line 295 - L306`) to `ShouldRender` method. https://github.com/ant-design-blazor/ant-design-blazor/blob/3ffafdf20033070d1793435cd02b6f1ca09a3ed1/components/table/Table.razor.cs#L295-L306

Reply:
@Zonciu As requested, commit 7a3b1e9. It definitely makes more sense to transfer all reloads before render is done. I still fill a bit intimidated by `Table` code...

Reply:
@ElderJames The idea here is to optimize by minimize number of renderings - running this code `OnAfterRender` means all is already rendered, so I would say that perf optimization of number of renderings in that event is a bit late.

Reply:
I see. You have made a great discovery.